Lib/test/test_venv.py

```diff
@@ -537,6 +537,69 @@ def test_pathsep_error(self):
         self.assertRaises(ValueError, venv.create, bad_itempath)
         self.assertRaises(ValueError, venv.create, pathlib.Path(bad_itempath))
 
+    @unittest.skipIf(os.name == 'nt', 'not relevant on Windows')
+    @requireVenvCreate
+    def test_zippath_from_non_installed_posix(self):
+        """
+        Test that when create venv from non-installed python, the zip path
+        value is as expected.
+        """
+        rmtree(self.env_dir)
+        # First try to create a non-installed python. It's not a real full
+        # functional non-installed python, but enough for this test.
+        non_installed_dir = os.path.realpath(tempfile.mkdtemp())
+        try:
```

Comment:
Rather than an enormous `try` block, you can use `self.addCleanup(rmtree, non_installed_dir)`

Reply:
Thanks. Do I need to wait for the buildbots test to finish before pushing the change?

Reply:
Yeah, let's get the failures fixed and clean up afterwards :)